crates/hyperswitch_connectors/src/connectors/fiuu/transformers.rs

```diff
@@ -1,226 +1,837 @@
-use masking::Secret;
+use masking::{PeekInterface, Secret};
 use serde::{Deserialize, Serialize};
+use strum::Display;
 
 use crate::{
-    types::{RefundsResponseRouterData, ResponseRouterData},
-    utils::PaymentsAuthorizeRequestData,
+    types::{
+        PaymentsCancelResponseRouterData, PaymentsCaptureResponseRouterData,
+        PaymentsSyncResponseRouterData, RefundsResponseRouterData, ResponseRouterData,
+    },
+    utils::{PaymentsAuthorizeRequestData, RouterData as _},
 };
 
-//TODO: Fill the struct with respective fields
 pub struct FiuuRouterData<T> {
-    pub amount: StringMinorUnit, // The type of amount that a connector accepts, for example, String, i64, f64, etc.
+    pub amount: StringMajorUnit,
     pub router_data: T,
 }
 
-impl<T> From<(StringMinorUnit, T)> for FiuuRouterData<T> {
-    fn from((amount, item): (StringMinorUnit, T)) -> Self {
-        //Todo :  use utils to convert the amount to the type of amount that a connector accepts
+impl<T> From<(StringMajorUnit, T)> for FiuuRouterData<T> {
+    fn from((amount, item): (StringMajorUnit, T)) -> Self {
         Self {
             amount,
             router_data: item,
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, PartialEq)]
+pub struct FiuuAuthType {
+    pub(super) merchant_id: Secret<String>,
+    pub(super) verify_key: Secret<String>,
+    pub(super) secret_key: Secret<String>,
+}
+
+impl TryFrom<&ConnectorAuthType> for FiuuAuthType {
+    type Error = Report<errors::ConnectorError>;
+    fn try_from(auth_type: &ConnectorAuthType) -> Result<Self, Self::Error> {
+        match auth_type {
+            ConnectorAuthType::SignatureKey {
+                api_key,
+                key1,
+                api_secret,
+            } => Ok(Self {
+                merchant_id: key1.to_owned(),
+                verify_key: api_key.to_owned(),
+                secret_key: api_secret.to_owned(),
+            }),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "UPPERCASE")]
+enum TxnType {
+    Sals,
+    Auts,
+}
+
+impl TryFrom<Option<CaptureMethod>> for TxnType {
+    type Error = Report<errors::ConnectorError>;
+    fn try_from(capture_method: Option<CaptureMethod>) -> Result<Self, Self::Error> {
+        match capture_method {
+            Some(CaptureMethod::Automatic) => Ok(Self::Sals),
+            Some(CaptureMethod::Manual) => Ok(Self::Auts),
+            _ => Err(errors::ConnectorError::CaptureMethodNotSupported.into()),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "UPPERCASE")]
+enum TxnChannel {
+    Creditz,
+}
+
+#[derive(Serialize, Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
 pub struct FiuuPaymentsRequest {
-    amount: StringMinorUnit,
-    card: FiuuCard,
+    #[serde(rename = "MerchantID")]
+    merchant_id: String,
```

Comment:
```suggestion
    merchant_id: Secret<String>,
```

Reply:
Please update any auth related details to Secret

Reply:
resolved